multihost_test/Bugzillas/test_bz_automation.py

```diff
@@ -146,3 +146,57 @@ def test_bz787736(self, multihost, create_backup):
         execute_cmd(multihost, "pwck -r > /tmp/anuj")
         assert "no matching password file entry" not in \
                execute_cmd(multihost, "cat /tmp/anuj").stdout_text
+
+    def test_bz_667593(self, multihost):
+        """
+        :title: Shadow-Utils: sg works with password
+         protected group with correct password
+        :id: 8e5d5324-4e23-11ed-95b7-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=667593
+        :steps:
+          1. Add user
+          2. Add group
+          3. Add password to group
+          4. Try good password with sg
+        :expectedresults:
+          1. Should succeed
+          2. Should succeed
+          3. Should succeed
+          4. Should succeed
+        """
+        # Adding user
+        execute_cmd(multihost, "useradd  tuser0011")
+        # Adding group
+        execute_cmd(multihost, "groupadd tgroup00011")
+        # script
+        file_location = "/multihost_test/Bugzillas/data/"
+        for file in [f'{file_location}bz_667593_1.sh',
+                     f'{file_location}bz_667593_4.sh',
+                     f'{file_location}add_passwod_to_group.sh']:
+            multihost.client[0].transport.put_file(os.getcwd()+f'/{file}', f'/tmp/{file}')
+            execute_cmd(multihost, f"chmod 755 /tmp/{file}")
+        # Adding password to group
+        execute_cmd(multihost, "sh /tmp/add_passwod_to_group.sh tgroup00011 Secret123")
+        # Trying good password with sg
+        cmd = execute_cmd(multihost, "sh /tmp/bz_667593_1.sh tuser0011 Secret123")
```

Comment:
Replace by `add_passwod_to_group.sh`

Reply:
add_passwod_to_group.sh takes input as:

$1 - group
$2 - password

Reply:
bz_667593.sh is replaced by add_password_to_group.sh